router/batchrouter/asyncdestinationmanager/common/common.go

```diff
@@ -137,10 +165,32 @@ type GetUploadStatsResponse struct {
 	Error      string        `json:"error"`
 }
 
+type TransformationFunc func(payload stdjson.RawMessage) string
+
 func GetTransformedData(payload stdjson.RawMessage) string {
 	return gjson.Get(string(payload), "body.JSON").String()
 }
 
+func GetMarshalledData(job *jobsdb.JobT, transformFn ...TransformationFunc) string {
+	fn := GetTransformedData
+	if len(transformFn) > 0 {
+		fn = transformFn[0]
+	}
+	payload := fn(job.EventPayload)
+	var asyncJob AsyncJob
+	err := stdjson.Unmarshal([]byte(payload), &asyncJob.Message)
+	if err != nil {
+		panic("Unmarshalling Transformer Response Failed")
+	}
+	asyncJob.Metadata = make(map[string]interface{})
+	asyncJob.Metadata["job_id"] = job.JobID
+	responsePayload, err := stdjson.Marshal(asyncJob)
+	if err != nil {
+		panic("Marshalling Response Payload Failed")
```

Comment:
can we propagate the error instead of panic here ?

Reply:
addressed.